python/whylogs/api/logger/experimental/logger/actor/pipe_signaler.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import threading as th
+from concurrent.futures import Future
+from queue import Empty
+from typing import Any, Dict, Generic, List, Optional, Tuple, TypeVar
+
+from whylogs.api.logger.experimental.logger.actor.actor import QueueConfig
+from whylogs.api.logger.experimental.logger.actor.faster_fifo_queue_wrapper import (
+    FasterQueueWrapper,
+)
+
+T = TypeVar("T")
+
+
+class PipeSignaler(th.Thread, Generic[T]):
+    """
+    ... [Your existing docstring] ...
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.daemon = True
+        self._logger = logging.getLogger(__name__)
+        self._queue_config = QueueConfig()
+        self.queue: FasterQueueWrapper[Tuple[str, Optional[Exception], Optional[T]]] = FasterQueueWrapper(
+            self._queue_config
+        )
+        self.futures: Dict[str, "Future[Any]"] = {}
+        self._end_polling = th.Event()
+        self._done = th.Event()
+
+    def signal(self, result: Tuple[str, Optional[Exception], Optional[T]]) -> None:
+        """
+        ... [Your existing docstring] ...
+        """
+        self.queue.send(result)
```

Comment:
these doc strings were removed by chat gpt. Can you sub them back in from the old code.